test/integration/link_dom.cc

```diff
@@ -358,6 +358,31 @@ TEST(DOMLink, Sensors)
   ASSERT_NE(nullptr, thermalCamSensor);
   EXPECT_EQ("my_thermal_camera", thermalCamSensor->Name());
 
+  // Get the segmentation sensor
+  const sdf::Sensor *segmentationSensor =
+    link->SensorByName("segmentation_sensor");
+  ASSERT_NE(nullptr, segmentationSensor);
+  EXPECT_EQ("segmentation_sensor", segmentationSensor->Name());
+  EXPECT_EQ(sdf::SensorType::SEGMENTATION_CAMERA, segmentationSensor->Type());
+  EXPECT_EQ(ignition::math::Pose3d(37, 38, 39, 0, 0, 0),
+            segmentationSensor->RawPose());
+  const sdf::Camera *segmentationCameraSensor =
+    segmentationSensor->CameraSensor();
+  ASSERT_NE(nullptr, segmentationCameraSensor);
+  EXPECT_EQ("my_segmentation_camera", segmentationCameraSensor->Name());
```

Comment:
```suggestion
  EXPECT_EQ("my_segmentation_camera", segmentationCameraSensor->Name());
  EXPECT_TRUE(segmentationCameraSensor->HasSegmentationType());
  EXPECT_EQ("semantic", segmentationCameraSensor->SegmentationType());
```
Now that the segmentation type is a part of the `sdf::Camera` API, can you add a check for the camera's segmentation type?